glommio/src/read_write_lock.rs

```diff
@@ -112,6 +114,18 @@ impl From<LockClosedError> for TryReadWriteLockError {
     }
 }
 
+impl From<LockClosedError> for io::Error {
+    fn from(er: LockClosedError) -> Self {
```

Comment:
I think it is useful to map - whenever possible - particular error kinds to specific `io::Error` error kinds.
This allows the user to test for what happened a lot easier in a call chain that is removed many layers from the first caller.

When we convert, I'd lean towards converting `Closed` to `BrokenPipe` and `WouldSuspend` (I like the name!) to `WouldBlock`. This will also make the conversion of channels easier.

Bringing @matklad into this since he had opinions about the errors as well.

Reply:
Well, I suppose in general error processing system is quite raw in Rust, yet. I suppose they already understood that and created error handling working group. If that would be an application I would suggest to use one of error processing creates which enhance embedded rust functionality. I like `anyhow` BTW. That is a minor fix, will do that ASAP.